### DISCUSSION TITLE
JEP issues - Java Embebeded Python

### DISCUSSION THREAD

**Discussion by rogerioubiratan** [sepinf-inc/IPED#842] [Q&A]
Boa tarde, estou com esse problema, ja fiz de tudo e não consegui resolver, estou usando o Ubuntu 20.10, java 8, versão build 1.8.0_202-b08, versão IPED 4.0.0, e da a mensagem:  2021-11-22 16:49:43	[ERROR]	[sepinf.python.PythonParser]			JEP não encontrado, todos os módulos python foram desligados. Veja https://github.com/sepinf-inc/IPED/wiki/User-Manual#python-modules
Gostaria de saber também qual prejuizo posso ter não obtendo esses modulos carregados. obrigado de antemão.

**Comment by lfcnassif** [comment]
What step didn't work for you? JEP installation on Linux should be pretty straightforward, it is more problematic on Windows.

Current features using Python/JEP are listed in the section of the manual you've liked above.

**Comment by lfcnassif** [comment]
Moving to discussions.

**Comment by rogerioubiratan** [comment]
I attach the processing log file:

[https://drive.google.com/file/d/14aaJwE2kwoZfk5yvEO_iER7a91UP0HDA/view?usp=sharing](url)


With some changes I use this standardization:

[https://drive.google.com/file/d/1-Ak59yins9OHifrrs3ygiRjsSUNPwCBa/view?usp=sharing](url)

**Reply by lfcnassif** [reply]
I can't access any of those links. Anyway, I meant what was the JEP installation step in the manual that failed for you.

**Comment by wladimirleite** [comment]
> I can't access any of those links. Anyway, I meant what was the JEP installation step in the manual that failed for you.

@lfcnassif, I think if you copy and paste the links URLs (instead of clicking) they should work. 
I took a look at the log, but I couldn't figure out anything useful, other than "JEP was not found"...

**Comment by rogerioubiratan** [comment]
https://drive.google.com/file/d/14aaJwE2kwoZfk5yvEO_iER7a91UP0HDA/view?usp=sharing

https://drive.google.com/file/d/1-Ak59yins9OHifrrs3ygiRjsSUNPwCBa/view?usp=sharing

**Comment by lfcnassif** [comment]
> https://drive.google.com/file/d/1-Ak59yins9OHifrrs3ygiRjsSUNPwCBa/view?usp=sharing

There is almost no output of the commands here, so it's impossible to know what is going on in your environment. What is the output of `pip install jep==3.9.1`?

PS: I saw you are trying to use libewf default repository, which is experimental and has an API incompatible with sleuthkit. I recommend you using libewf-legacy, as recommended by libewf author Joachim Metz.

**Comment by rogerioubiratan** [comment]
root@iped:/home/iped/Iped/IPED/target/release/iped-4.0.0-snapshot# pip install jep==3.9.1
Requirement already satisfied: jep==3.9.1 in /usr/local/lib/python3.9/dist-packages (3.9.1)
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv

**Reply by lfcnassif** [reply]
Have you followed step below in the manual? I can't find this in your commands.

> After you have compiled JEP successfully, you need to include jep.dll parent folder to PATH environment variable on Windows (usually located in C:\Users%user%\AppData\Local\Programs\Python\Python38\Lib\site-packages\jep) or jep.so parent folder to LD_LIBRARY_PATH env var on Linux.

**Comment by rogerioubiratan** [comment]
https://drive.google.com/file/d/10_Al8119t3W8SESUOFfLIjhX6Qy_x50B/view?usp=sharing

commands executed:

**Reply by lfcnassif** [reply]
> export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/local/lib/python3.9/dist-packages/jep/libjep.so

This is not what was pointed out. I also saw different versions of JEP being installed one over another (the repository default, 3.9.1, and from github JEP master branch). You installed python 3.9, the manual recommends python 3.8. I recommend you uninstalling every version of JEP and python in your system and making a clean install of the recommended versions again, following the manual. Also please post the output of all your commands.

**Comment by rogerioubiratan** [comment]
nothing yet, I put the command above and it continues with the message:
root@iped:/home/iped/Iped/IPED/target/release/iped-4.0.0-snapshot# java -jar iped.jar -d /media/iped/Rogerio1TB/imagens/ALVO104-ITEM05.4/ALVO06-ITEM05.4.E01 -o /media/iped/Rogerio1TB/teste100
2021-11-23 14:21:12	[ERROR]	[indexer.util.LibreOfficeFinder]			LibreOffice 7 not tested!
2021-11-23 14:21:13	[ERROR]	[sepinf.python.PythonParser]			JEP não encontrado, todos os módulos python foram desligados. Veja https://github.com/sepinf-inc/IPED/wiki/User-Manual#python-modules
Temp Folder for Libraries: /tmp/indexador-temp-8668815831612232140
SleuthkitJNI: loaded libtsk_jni

Processing another acquisition now.

**Comment by rogerioubiratan** [comment]
I'm going to reformat the computer and start from scratch, I've tried it in Caine, but there are a lot of errors, so continue and solve it in Ubuntu, then I'll give you feedback, and thanks for now.

**Comment by aberenguel** [comment]
I installed JEP in Ubuntu 22.04 using this command:

```shell
python3 -m pip install jep==3.9.1
```

Sometimes the java does not find `libjep.so`, so I set the library path in this way:

```shell
export LD_LIBRARY_PATH=$HOME/.local/lib/python3.10/site-packages/jep/
```

**Reply by afonsohca** [reply]
> Instalei o JEP no Ubuntu 22.04 usando este comando:
> 
> ```shell
> python3 -m pip install jep==3.9.1
> ```
> 
> Às vezes o java não encontra `libjep.so`, então eu definai o caminho da biblioteca desta forma:
> 
> ```shell
> export LD_LIBRARY_PATH=$HOME/.local/lib/python3.10/site-packages/jep/
> ```

Estou tendo o mesmo problema. Meu java não encontrou o libjep.so, mas, vi no diretório que não foi criado o arquivo em questão...não existe um artefado libjep.so.

**Reply by lfcnassif** [reply]
You should find where the (lib)jep.so artifact was created in your system. If it wasn't, I suggest you asking for help from JEP project.

**Comment by afonsohca** [comment]
Veja essa saída:
![image](https://user-images.githubusercontent.com/110132252/182883240-53be6eac-1bf7-4711-92a4-ba2f58ba0a45.png)

E o log:
![image](https://user-images.githubusercontent.com/110132252/182883364-e3f77914-a05a-457a-bf4d-dabdbcb89b48.png)

**Comment by aberenguel** [comment]
Try to install and execute the commands without `sudo`.
I think if you run java with sudo, it search the jep.jar in `/root/.local/lib/python3.10/site-packages/jep/` ou `/usr/local/`
